### PR TITLE
Fix unnecessary whole project recompilation

### PR DESCRIPTION
**Pull request by volodg**
closes https://github.com/CocoaPods/CocoaPods/issues/7187

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix unnecessary the whole project recompilation  
  [volodg](https://github.com/volodg)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Fix unnecessary whole project recompilation with static frameworks
+[volodg](https://github.com/volodg)
+[#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```
Comment (orta) [inline CHANGELOG.md]: Your issue number is https://github.com/CocoaPods/CocoaPods/issues/7187 - thanks 👍

**Comment by dnkoutso** [comment]
Please squash into a single commit.

**Comment by volodg** [comment]
Done

**Comment by dnkoutso** [comment]
Change looks good to me. I'd still like @paulb777 to weigh in on this before we merge.

**Comment by paulb777** [comment]
Plan to review by end of day Thursday

On Nov 1, 2017 17:41, "Dimitris Koutsogiorgas" <notifications@github.com>
wrote:

Change looks good to me. I'd still like @paulb777
<https://github.com/paulb777> to weigh in on this before we merge.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/CocoaPods/CocoaPods/pull/7190#issuecomment-341284936>,
or mute the thread

**Comment by dnkoutso** [comment]
Need a rebase.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by volodg** [comment]
rebased

**Comment by SuperQiurongchao** [comment]
My pod version is 1.6.1. Now my cocoapod use static libraries，and my project always recompile when I build it

**Pull request merged**